src/Meziantou.Analyzer/Rules/ClassWithEqualsTShouldImplementIEquatableTFixer.cs

```diff
@@ -47,10 +48,20 @@ private static async Task<Document> ImplementIEquatable(Document document, Synta
             if (genericInterfaceSymbol is null)
                 return document;
 
-            var concreteInterfaceSymbol = genericInterfaceSymbol.Construct(declaredTypeSymbol);
-            if (concreteInterfaceSymbol is null)
+            // Retrieve Nullable Annotation from the Equals method and use it to construct the concrete interface
```

Comment:
- I'm not sure we need to add the nullable annotation because `IEquatable<T>.Equals` [has the `[AllowNull]` attribute](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/IEquatable.cs#L9). The dotnet/runtime repository seems to contain more examples without the nullable annotation than with the nullable annotation. But maybe this is because it is mostly old code. At the same time I think this is clearer to have the annotation. => I really don't have a strong opinion on it, so I think we can go with the nullable annotation as you have already implemented it.

- Instead of doing the logic in the analyzer and the fixer to find the Equals method, you should do it once. You can use an internal static method in the analyzer or store data using the properties dictionary.

Reply:
Done